scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Connection.java

```diff
@@ -83,6 +94,8 @@ public Connection(final InetSocketAddress peerAddress, final Long autoResumption
 			this.peerAddress = peerAddress;
 			this.autoResumptionTimeout = autoResumptionTimeout;
 			this.sessionListener = new ConnectionSessionListener();
+			this.maxDeferredProcessedApplicationDataMessages = maxDeferredProcessedApplicationData;
+			this.deferredApplicationData = new ArrayList<Object>(maxDeferredProcessedApplicationData);
```

Comment:
Why not? For the outgoing connections, this will always hold the outgoing messages. So only incoming may save a little, if no app data is received out of order.

Reply:
I would say why not saving this "little data" ? to save a "little of CPU" ? I don't know what is better.

By the way if someone configure `maxDeferredProcessedApplicationDataMessages` to a bigger number this could be more than just "little data" but ok choosing a big value for `maxDeferredProcessedApplicationDataMessages` is probably stupid...

Anyway, let it like this if you find this better.